magicctapipe/scripts/lst1_magic/lst1_magic_dl2_to_dl3.py

```diff
@@ -354,8 +365,152 @@ def create_pointing_table(event_table):
+            ghcuts=hdus_irf["GH_CUTS"] 
+            ghcuts_low=u.Quantity(ghcuts.data['low'], unit=ghcuts.columns['low'].unit)
+            ghcuts_high=u.Quantity(ghcuts.data['high'], unit=ghcuts.columns['high'].unit)
+            ghcuts_center=u.Quantity(ghcuts.data['center'], unit=ghcuts.columns['center'].unit)
+            if ghcuts_low_last is not None:
+                if (ghcuts_low_last!=ghcuts_low).any() or (ghcuts_high_last!=ghcuts_high).any() or (ghcuts_center_last!=ghcuts_center).any():
+                    raise ValueError('Non matching bins in GH_CUTS')                    
+                        
+            ghcuts_all.append(ghcuts.data['cut'])
+            
+            ghcuts_low_last=ghcuts_low
+            ghcuts_high_last=ghcuts_high
+            ghcuts_center_last=ghcuts_center
+        else:
+            logger.warning("skipping "+ name)
+
+    # fix me! only the last file is checked, no consistency checks
+    extra_headers={key: aeff.header[key] for key in list(set(extra_headers_list) & set(aeff.header.keys()))}    
+
+    points=np.array(points)
+    aeff_energ_bins=read_fits_bins_lo_hi(hdus_irfs, aeff_ext_name, "ENERG")
+    aeff_theta_bins=read_fits_bins_lo_hi(hdus_irfs, aeff_ext_name, "THETA")
+    
+    edisp_energ_bins=read_fits_bins_lo_hi(hdus_irfs, edisp_ext_name, "ENERG")
+    edisp_migra_bins=read_fits_bins_lo_hi(hdus_irfs, edisp_ext_name, "MIGRA")
+    edisp_theta_bins=read_fits_bins_lo_hi(hdus_irfs, edisp_ext_name, "THETA")
+    
+    
+    aeff_all= u.Quantity(np.array(aeff_all), hdus_irf[aeff_ext_name].columns["EFFAREA"].unit, copy=False)
+    edisp_all= u.Quantity(np.array(edisp_all), hdus_irf[edisp_ext_name].columns["MATRIX"].unit, copy=False)
+    ghcuts_all=np.array(ghcuts_all)
+    
+    # now read the file and check for what to interpolate
+    data=pd.read_hdf(input_file_dl2, 'events/parameters')
+    data_coszd = np.mean(np.sin(data['pointing_alt']))
+    data_az = np.mean(data['pointing_az'])
+    
+    target=np.array([data_coszd, data_az])
+    logger.info("interpolate for: "+str(target))
+    aeff_interp=interp.interpolate_effective_area_per_energy_and_fov (aeff_all, points, target, method=method)
+    
+    # here we need to swap axes because the function expects shape: (n_grid_points, n_energy_bins, n_migration_bins, n_fov_offset_bins)
+    edisp_interp=interp.interpolate_energy_dispersion(np.swapaxes(edisp_all, 1, 3), points, target, method=method)
+    
+    # now create the HDUS
+    # to have the same format we need to loose one dimention, this might need to be rewised for files with many bins in offset angle!
+    aeff_hdu=create_aeff2d_hdu(aeff_interp[...,0], aeff_energ_bins[0], aeff_theta_bins[0], extname=aeff_ext_name, **extra_headers)
+    edisp_hdu=create_energy_dispersion_hdu(edisp_interp[0,...], edisp_energ_bins[0], edisp_migra_bins[0], edisp_theta_bins[0], extname=edisp_ext_name)
+    
+    ghcuts_interp = griddata(points, ghcuts_all, target, method=method)
```

Comment:
there is no function for interpolation of cut values in pyirf yet, so I implemented it directly here for the time being using the same interpolation engine as Aeff and edisp above. 
Note that if we run with variable theta2 cut it would also need to be interpolated (not done yet)

Reply:
Thanks, I also implemented the interpolation for the theta (= rad_max) cuts.